### PR TITLE
feat: validate building status when changing building outline

### PR DESCRIPTION
**Pull request by jvandaal**

### CODE REVIEW

**Review by ArneD — APPROVED** [review]

**Review by ArneD — APPROVED** [review]

File: src/BuildingRegistry/Building/Building.cs [hunk 1]
```diff
@@ -56,6 +56,17 @@ public void ChangeOutlining(ExtendedWkbGeometry extendedWkbGeometry)
         {
             GuardRemovedBuilding();
 
+            var invalidStatuses = new List<BuildingStatus>
```
Comment (lvermeulen) [inline src/BuildingRegistry/Building/Building.cs:59]: should be initialized once (static readonly)

File: test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs [hunk 2]
```diff
@@ -51,7 +51,7 @@ public void ThenThrowValidationException()
                 .Result
                 .Where(x => x.Errors.Any(e =>
                     e.ErrorCode == "GebouwStatusNietInGeplandOfGerealiseerd"
-                    && e.ErrorMessage == "Deze actie is enkel toegestaan binnen een gepland of gerealiseerd gebouw."));
+                    && e.ErrorMessage == "Deze actie is enkel toegestaan binnen een gepland of gerealiseerd gebouw."));
```
Comment (lvermeulen) [inline test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs:54]: indent

File: src/BuildingRegistry/Building/Building.cs (hunk 1 above)
Comment (jvandaal) [reply src/BuildingRegistry/Building/Building.cs]: Moved to `private static readonly List<BuildingStatus> ChangeBuildingOutlineInvalidStatuses` member

File: test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs (hunk 2 above)
Comment (jvandaal) [reply test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs:54]: In my editor the indentation seems correct? 
image

Comment (lvermeulen) [reply test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs:54]: the && should be indented one level more

Comment (jvandaal) [reply test/BuildingRegistry.Tests/BackOffice/Api/WhenCorrectingBuildingUnitPosition/GivenBuildingHasInvalidStatusException.cs]: Done

**Review by lvermeulen — APPROVED** [review]

**Pull request merged**